source/common/http/http2/codec_impl.h

```diff
@@ -84,6 +84,17 @@ class ConnectionImpl : public virtual Connection, Logger::Loggable<Logger::Id::h
   Protocol protocol() override { return Protocol::Http2; }
   void shutdownNotice() override;
   bool wantsToWrite() override { return nghttp2_session_want_write(session_); }
+  // Propogate network connection watermark events to each stream on the connection.
+  void onUnderlyingConnectionAboveWriteBufferHighWatermark() override {
```

Comment:
Just for a public record of our IM conversation, it is possible that we kill onUnderlying* functions entirely and just have codec also register for connection callbacks. I don't feel strongly about it one way or the other (though I think it might simplify code further), but just pointing it out.